### PR TITLE
`env_sky` rendering

### PR DESCRIPTION
**Pull request by kristiker**
2D skybox rendering.
image
image

### CODE REVIEW

File: GUI/Types/Renderer/MaterialLoader.cs [hunk 1]
```diff
@@ -88,7 +88,14 @@ bool TryBindTexture(RenderMaterial mat, string name, string path)
             {
                 if (mat.Shader.GetUniformLocation(name) != -1)
                 {
-                    mat.Textures[name] = LoadTexture(path);
+                    if (name != "g_tSkyTexture")
```
Comment (xPaw) [inline GUI/Types/Renderer/MaterialLoader.cs:91]: Is checking name going to work for cubemap entities too?

File: GUI/Types/Renderer/GLSceneViewer.cs [hunk 2]
```diff
@@ -233,6 +235,12 @@ protected void SetEnabledLayers(HashSet<string> layers)
         private void SetRenderMode(string renderMode)
         {
             ViewerControl.Camera?.Picker.SetRenderMode(renderMode);
+            if (Scene.Sky != null)
```
[the remark this answers is not in the document]
Comment (kristiker) [reply GUI/Types/Renderer/GLSceneViewer.cs:238]: It is. This reloads the material, useful when iterating the shader.

File: GUI/Types/Renderer/MaterialLoader.cs (hunk 1 above)
Comment (kristiker) [reply GUI/Types/Renderer/MaterialLoader.cs:91]: I don't plan to hardcode this. I think I will end up adding (targetType, handle) as values to the Textures dictionary.

File: GUI/Types/Renderer/ShaderLoader.cs [hunk 3]
```diff
@@ -172,7 +176,7 @@ private static string UpdateDefines(string source, IDictionary<string, bool> arg
                 var defaultValue = define.Groups["DefaultValue"];
                 var index = defaultValue.Index;
                 var length = defaultValue.Length;
-                var newValue = value ? "1" : "0";
+                var newValue = $"{value}";
```
Comment (xPaw) [inline GUI/Types/Renderer/ShaderLoader.cs:179]: value.ToString()

File: GUI/Types/Renderer/ShaderLoader.cs [hunk 4]
```diff
@@ -288,8 +294,8 @@ private uint CalculateShaderCacheHash(string shaderFileName, IDictionary<string,
 
             foreach (var key in parameters)
             {
-                shaderCacheHashString.AppendLine(key);
-                shaderCacheHashString.AppendLine(arguments[key] ? "t" : "f");
+                shaderCacheHashString.Append(key);
+                shaderCacheHashString.Append(Convert.ToBase64String(new[] { arguments[key] }));
```
Comment (xPaw) [inline GUI/Types/Renderer/ShaderLoader.cs:298]: Isn't it a byte, just doing ToString should be enough on it?

Comment (kristiker) [reply GUI/Types/Renderer/ShaderLoader.cs:298]: Murmurhash didn't like numbers, this fixed the collisions. Presumably that's why it used to have "t" and "f".

Comment (kristiker) [reply GUI/Types/Renderer/ShaderLoader.cs]: Nevermind ToString works fine now 😄

**Review by xPaw — APPROVED** [review]

**Pull request merged**